Gems/LyShine/Code/Source/UiTextInputComponent.cpp

```diff
@@ -332,6 +337,14 @@ bool UiTextInputComponent::HandleTextInput(const AZStd::string& inputTextUTF8)
 
     bool changedText = false;
 
+#if defined(CARBONATED) && defined(AZ_PLATFORM_IOS)
```

Comment:
O3DE (rightfully) doesn't like platform-specific code in platform-agnostic engine code.  The UiTextInputComponent being a good example - it should not care about ios or android or whatever.  You'll need to remove the AZ_PLATFORM_IOS guard here and make this code platform agnostic.

Reply:
Removed all the AZ_PLATFORM defines in UiTextInputComponent